rules/DowngradePhp80/Rector/FuncCall/DowngradeStrContainsRector.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\DowngradePhp80\Rector\FuncCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\BinaryOp\Identical;
+use PhpParser\Node\Expr\BinaryOp\NotIdentical;
+use PhpParser\Node\Expr\BooleanNot;
+use PhpParser\Node\Expr\FuncCall;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @changelog https://wiki.php.net/rfc/str_contains
+ *
+ * @see Rector\Tests\DowngradePhp80\Rector\FuncCall\DowngradeStrContainsRector\DowngradeStrContainsRectorTest
+ */
+final class DowngradeStrContainsRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Replace str_contains() with strpos() !== false',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run()
+    {
+        return str_contains('abc', 'a');
+    }
+}
+CODE_SAMPLE
+,
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run()
+    {
+        return strpos('abc', 'a') !== false;
+    }
+}
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [FuncCall::class, BooleanNot::class];
+    }
+
+    /**
+     * @param FuncCall|BooleanNot $node
+     * @return Identical|NotIdentical|null The refactored node.
+     */
+    public function refactor(Node $node): Identical | NotIdentical | null
+    {
+        $funcCall = $this->matchStrContainsOrNotStrContains($node);
+
+        if (! $funcCall instanceof FuncCall) {
+            return null;
+        }
+
+        /** @var string $haystack */
+        $haystack = $funcCall->args[0]->value;
+        /** @var string $needle */
+        $needle = $funcCall->args[1]->value;
```

Comment:
Here can be anything, not only `string`. The annotations are no needed probably

Reply:
See 8410f12